File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        splits = 0
        rsum = 0
        lsum = 0
        
        for i in range(len(nums)):
            rsum = rsum + nums[i]

        for i in range(len(nums)-1):
            lsum = lsum + nums[i]
            rsum = rsum - nums[i]

            if(lsum>=rsum):
                splits += 1

        return splits

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int splits = 0;
        int n = nums.size();
        long long sumleft = 0, sumright = 0;

        for(auto c: nums){
            sumright += c;
        }

        for(int i=0; i<n-1; i++){
            sumleft += nums[i];
            sumright -= nums[i];

            if(sumleft >= sumright){
                splits++;
            }
        }
        return splits;
    }
};